lib/utils/grpc.go

```diff
@@ -20,16 +20,66 @@ import (
 	"context"
 
 	"github.com/gravitational/trace/trail"
+
 	"google.golang.org/grpc"
 )
 
-func ErrorConvertUnaryInterceptor(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+// grpcServerStreamWrapper wraps around the embedded grpc.ServerStream
+// and intercepts the RecvMsg and SendMsg method calls converting errors to the
+// appropriate grpc status error.
+type grpcServerStreamWrapper struct {
+	grpc.ServerStream
+}
+
+// SendMsg wraps around ServerStream.SendMsg and adds metrics reporting
+func (s *grpcServerStreamWrapper) SendMsg(m interface{}) error {
+	return trail.ToGRPC(s.ServerStream.SendMsg(m))
+}
+
+// RecvMsg wraps around ServerStream.RecvMsg and adds metrics reporting
+func (s *grpcServerStreamWrapper) RecvMsg(m interface{}) error {
+	return trail.ToGRPC(s.ServerStream.RecvMsg(m))
+}
```

Comment:
I'm under the (potentially wrong) impression that the errors returned by `SendMsg` and `RecvMsg` are errors being bubbled up from the GRPC library to the calling code (i.e. teleport), in which case you'd want to be using `trail.FromGRPC` if anything.  I got that impression based on poking around in some generated code that I don't fully understand, so take it with a grain of salt.

Either way, this definitely deserves some test coverage to verify that the `trace` error types are being preserved correctly across the wire.